nuxeo-core-binarymanager-s3/src/test/java/org/nuxeo/ecm/core/storage/sql/TestS3DirectBatchHandler.java

```diff
@@ -0,0 +1,198 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.ecm.core.storage.sql;
+
+import static org.junit.Assume.assumeTrue;
+
+import java.io.ByteArrayInputStream;
+import java.util.Map;
+import java.util.Random;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.Batch;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchHandler;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchManager;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.BatchFileInfo;
+import org.nuxeo.ecm.automation.test.AutomationFeature;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import com.amazonaws.SdkBaseException;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+
+@RunWith(FeaturesRunner.class)
+@Features(AutomationFeature.class)
+@Deploy({ "org.nuxeo.ecm.automation.io",
+        "org.nuxeo.ecm.automation.server",
+        "org.nuxeo.ecm.core.storage.binarymanager.s3" })
+public class TestS3DirectBatchHandler {
+
+    public static final int MULTIPART_THRESHOLD = 5 * 1024 * 1024; // 5MB AWS minimum value
+
+    public static final String S3DIRECT_PREIX = "nuxeo.s3storage.transient.";
+
+    private static String envId;
+
+    private static String envSecret;
+
+    @Inject
+    public BatchManager batchManager;
+
+    @BeforeClass
+    public static void beforeClass() {
+        envId = System.getenv(S3BinaryManager.AWS_ID_ENV);
+        envSecret = System.getenv(S3BinaryManager.AWS_SECRET_ENV);
+        assumeTrue("AWS Credentials not set in the environment variables", StringUtils.isNoneBlank(envId, envSecret));
+        System.setProperty(S3DIRECT_PREIX + S3BinaryManager.AWS_ID_PROPERTY, envId);
+        System.setProperty(S3DIRECT_PREIX + S3BinaryManager.AWS_SECRET_PROPERTY, envSecret);
```

Comment:
can't use an isolated properties object passed to binaryManager.initialize ?

Reply:
Actually, the environment variable values are set into system properties to simulate values from nuxeo.conf, there are used in a xml contribution in the methods annotated with @Deploy